### ISSUE TITLE
False positive: type checker thinks `__getitem__` is required for `foo[2] = 2`

### ISSUE THREAD

**Issue by mflova** [astral-sh/ty#862]
### Summary

It seems the tool mixes `__getitem__` protocol with `__setitem__`. In the situation below, `ty` seems to be thinking that I am accessing the `__getitem__` method when it is `__setitem__`:

```py
class Foo:
    def __setitem__(self, key: int, value: int) -> None:
        print(f"Setting {key} to {value}")


foo = Foo()
foo[2] = 2  # Cannot subscript object of type `Foo` with no `__getitem__` method
```

### Version

ty 0.0.1-alpha.15 (0369a3598 2025-07-18)

**Comment by carljm** [comment]
Thanks for the report! This is definitely a bug.

**Comment by MatthewMckee4** [comment]
Help wanted? If no one's has picked it up in the next couple of days I will when I'm available, thanks

**Comment by AlexWaygood** [comment]
go for it!
